// components/headerMenu/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import {browserHistory, withRouter} from 'react-router'
import {compose} from 'redux'

import IconButton from 'material-ui/IconButton';

import Menu, { MenuItem } from 'material-ui/Menu';
import AccountCircle from 'material-ui-icons/AccountCircle';
import {getUserSession, logout} from "actions";
import Button from 'material-ui/Button';
import {Link} from 'react-router'
class HeaderMenu extends  Component {
    constructor (props) {
        super(props);
        this.state = {
            anchor: 'left',
        };
    }
    handleMenu = event => {
        this.setState({ anchorEl: event.currentTarget });
    };
    handleClose = () => {
        this.setState({ anchorEl: null });
    };
    loginUser  = () =>{
        this.props.router.push('/');
    };
    logoutUser  = () =>{
        this.props.router.push('/');
    };

    render() {
        const { anchorEl } = this.state;
        const openL = Boolean(anchorEl);
        const isLogin = this.props.userSession.isLogin;
        if (isLogin){
            return (
                <div >
                    <IconButton
                        aria-owns={openL ? 'menu-appbar' : null}
                        aria-haspopup="true"
                        onClick={this.handleMenu}
                        color="inherit"
                    >
                        <AccountCircle />
                    </IconButton>
                    <Menu
                        id="menu-appbar"
                        anchorEl={anchorEl}
                        anchorOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        transformOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        open={openL}
                        onClose={this.handleClose}
                    >
                        <MenuItem >Profile</MenuItem>
                        <MenuItem onClick={() =>this.props.router.push('/')}>
                            LogOut
                        </MenuItem>
                    </Menu>
                </div>
            );
        } else {
            return (<Button  onClick={this.loginUser } color="inherit">Login</Button>);
        }

    }
};

const mapStateToProps = (store) => {
    return {
        userSession: store.userSession
    }
};

const mapDispatchToProps = {
    getUserSession,
    logout
};
export default compose(
    withRouter,
    connect(mapStateToProps, mapDispatchToProps)
)(HeaderMenu)
