// components/headerMenu/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import {withRouter} from 'react-router'
import {compose} from 'redux'

import IconButton from 'material-ui/IconButton';


import {withStyles} from "material-ui/styles/index";

import Menu, { MenuItem } from 'material-ui/Menu';
import AccountCircle from 'material-ui-icons/AccountCircle';
import {otherMailFolderListItems} from "components/mainPageLayout/tileData";
import {isUserLogin, toggleNavMenu} from "actions";
import Button from 'material-ui/Button';






class HeaderMenu extends  Component {
    constructor (props) {
        super(props);
        this.state = {
            anchor: 'left',
        };
    }


    handleMenu = event => {
        this.setState({ anchorEl: event.currentTarget });
    };
    handleClose = () => {
        this.setState({ anchorEl: null });
    };
    loginUser  = () =>{
        this.props.isUserLogin(true);
    };
    logoutUser = () =>{
        this.props.isUserLogin(false);
        this.setState({ anchorEl: null });

    };
    render() {
        const { anchorEl } = this.state;
        const openL = Boolean(anchorEl);
        const isLogin = this.props.userSession.isLogin;

        if (isLogin){
            return (
                <div >
                    <IconButton
                        aria-owns={openL ? 'menu-appbar' : null}
                        aria-haspopup="true"
                        onClick={this.handleMenu}
                        color="inherit"
                    >
                        <AccountCircle />
                    </IconButton>
                    <Menu
                        id="menu-appbar"
                        anchorEl={anchorEl}
                        anchorOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        transformOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        open={openL}
                        onClose={this.handleClose}
                    >
                        <MenuItem onClick={this.loginUser}>Profile</MenuItem>
                        <MenuItem onClick={this.logoutUser}>LogOut</MenuItem>
                    </Menu>
                </div>
            );
        } else {
            return (<Button  onClick={this.loginUser }color="inherit">Login</Button>);
        }

    }
};

const mapStateToProps = (store) => {
    return {
        userSession: store.userSession
    }
};

const mapDispatchToProps = {
    isUserLogin
};
export default compose(
    withRouter,
    connect(mapStateToProps, mapDispatchToProps)
)(HeaderMenu)
