// components/loginForm/index.js
import React, {Component} from 'react';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';
import {login} from "actions";
import {connect} from "react-redux";

class LoginForm extends Component  {

    constructor (props) {
        super(props);
        this.store = this.props.store;
        this.state = {
            userName: ''
        };
        this.loginUser = this.loginUser.bind(this);

    }

        loginUser(){
            this.props.login(
                {
                    email: this.userEmail.value,
                    pass: this.password.value
                }
            );

        }
    render(){
        return (
            <div>
                <TextField
                    inputRef={el => this.userEmail = el}
                    label="LogIn"
                    margin="normal"/>
                <br/>
                <TextField
                    inputRef={el => this.password = el}
                    label="Password"
                    id="margin-dense"
                    margin="normal"/>
                <br/>
                <Button onClick={this.loginUser} raised color="primary">
                    Log In
                </Button>

            </div>
        );
    }

}
const mapStateToProps = (state) => {
    return {
        users: state
    }
};

const mapDispatchToProps = {
    login
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginForm);
