// components/loginForm/index.js
import React, {Component} from 'react';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';
import { login } from "actions";
import {connect} from "react-redux";

class LoginForm extends Component  {

    constructor (props) {
        super(props);
        this.loginUser = this.loginUser.bind(this);
    }

        loginUser(){
            const email = this.email.value;
            const pass = this.password.value;
            this.props.login({email, pass});
        }

    render(){
        return (
            <div>
                <TextField
                    inputRef={e => this.email = e}
                    label="LogIn"
                    margin="none"/>
                <br/>
                <TextField
                    inputRef={e => this.password = e}
                    label="Password"
                    id="margin-dense"
                    margin="none"/>
                <br/>
                <Button  style={{margin:'20px 0px'}} onClick={this.loginUser} raised color="primary">
                    Log In
                </Button>

            </div>
        );
    }

}
const mapStateToProps = (state) => {
    return {
        users: state
    }
};

const mapDispatchToProps = {
    login
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginForm);
