// containers/dashboard/index.js
import React, {Component} from 'react'
import TextField from 'material-ui/TextField';
import {connect} from 'react-redux'
import Button from 'material-ui/Button';
import {
    fetchUsers, createUser, login
} from 'actions'
import UserTable from 'components/userTable'



class Dashboard extends Component {
    constructor (props) {
        super(props);

        this.store = this.props.store;
        this.state = {
            userName: ''
        };
        this.createUser = this.createUser.bind(this);
        this.loginUser = this.loginUser.bind(this);
    }

    loginUser(){
        this.props.login(
            {
                email: this.userEmail.value,
                pass: this.password.value
            }
        );

    }

    componentDidMount () {
        this.props.fetchUsers();
        console.log("this.props.store");
        console.log(this.props.store);
    }
    createUser(){
        console.log(this.name.value);
        this.props.createUser(
            {
                userName: this.name.value,
                email: this.email.value,
                pass: this.pass.value
            }
        );
        // this.props.fetchUsers();
        this.name.value = '';
        this.email.value = '';
        this.pass.value = '';
    }
    render () {
        return (
            <main className='dashboard-page'>
                <div className='col-md-6' >
                    <div className='row'>
                      <div className="col-md-12">
                        <TextField
                            inputRef={el => this.name = el}
                            label="User Name"
                            margin="normal"
                             />
                        <br/>
                        <TextField
                            inputRef={el => this.email = el}
                            label="Email"
                            id="margin-dense"
                            margin="normal"/>
                        <br/>
                        <TextField
                            inputRef={el => this.pass = el}
                            label="Password"
                            id="margin-normal"
                            margin="normal"/>
                         </div>
                    </div>
                        <Button  onClick={this.createUser} raised color="primary">
                        Create User
                         </Button>

                </div>
                <div className='col-md-6'>
                    <TextField
                        inputRef={el => this.userEmail = el}
                        label="LogIn"
                        margin="normal"/>
                    <br/>
                    <TextField
                        inputRef={el => this.password = el}
                        label="Password"
                        id="margin-dense"
                        margin="normal"/>
                    <br/>

                    <Button onClick={this.loginUser} raised color="primary">
                      Log In
                    </Button>

                </div>
                <div className="col-md-12">
                    <UserTable></UserTable>
                </div>
            </main>

        )
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        users: state
    }
};

const mapDispatchToProps = {
    fetchUsers,
    createUser,
    login

};

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
