// containers/dashboard/index.js
import React, {Component} from 'react'
import TextField from 'material-ui/TextField';
import {connect} from 'react-redux'
import Button from 'material-ui/Button';
import Icon from 'material-ui/Icon';
import {
    fetchUsers,createUser,login
} from 'actions'
import UserTable from 'components/userTable'

import Delete from 'material-ui-icons/Delete';
import FileUpload from 'material-ui-icons/FileUpload';
import KeyboardVoice from 'material-ui-icons/KeyboardVoice';

import Save from 'material-ui-icons/Save';

class Dashboard extends Component {
    constructor (props) {
        super(props)
        this.state = {
            userName: ''
        };
        this.сreateUser = this.сreateUser.bind(this);
        this.loginUser = this.loginUser.bind(this);
    }

    loginUser(){

        console.log("LogIn");
        this.props.login(
            {
                email: this.userEmail,
                pass: this.password
            }
        );

    }

    componentDidMount () {
        this.props.fetchUsers();
    }
    сreateUser(){
        console.log(this.name.value);

        this.props.createUser(
            {
                userName: this.name.value,
                email: this.email.value,
                pass: this.pass.value
            }
        );
        this.name.value = '';
        this.email.value = '';
        this.pass.value = '';
    }
    render () {
        const { classes } = "sadasdas";
        return (
            <main className='dashboard-page'>

                <div className='col-md-6' >
                    <div class="col-md-12">
                        <TextField
                            inputRef = {el => this.name = el}
                            label="User Name"
                            margin="normal"
                             />
                        <br/>
                        <TextField
                            inputRef={el => this.email = el}
                            label="Email"
                            id="margin-dense"
                            margin="normal"/>
                        <br/>
                        <TextField
                            inputRef={el => this.pass = el}
                            label="Password"
                            id="margin-normal"
                            margin="normal"/>
                         </div>
                        <Button  onClick={this.сreateUser} raised color="primary">
                        Create User
                         </Button>

                </div>
                <div className='col-md-6'>
                    <TextField
                        inputRef = {el => this.userEmail = el}
                        label="LogIn"
                        id="margin-none"/>
                    <br/>
                    <TextField
                        inputRef={el => this.password = el}
                        label="Password"
                        id="margin-dense"
                        margin="dense"/>
                    <br/>

                    <Button onClick={this.loginUser} raised color="primary">
                      Log In
                    </Button>

                </div>
                <div className="col-md-12">
                    <UserTable></UserTable>
                </div>
            </main>

        )
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        state: state,
        ownProps: ownProps
    }
};

const mapDispatchToProps = {
    fetchUsers,
    createUser,
    login

};

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard)
