// components/navigationMenu/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import {withRouter} from 'react-router'
import {compose} from 'redux'

import Drawer from 'material-ui/Drawer';
import Button from 'material-ui/Button';
import List from 'material-ui/List';

import { mailFolderListItems } from './tileData';
import './main.scss';

class NavigationMenu extends Component {
    constructor (props) {
        super(props)
        this.state = {
            top: false,
            left: false,
            bottom: false,
            right: false,
        };
    }
    componentDidMount(){
    }
    toggleDrawer = (side, open) => () => {
        this.setState({
            [side]: open,
        });
    };

    render() {
        const sideList = (
            <div >
                <List>{mailFolderListItems}</List>
            </div>
        );
        return (
            <div >
                <Button onClick={this.toggleDrawer('left', true)}>Open Left</Button>
                <Drawer className={'main-navigation-menu'} open={this.state.left} onClose={this.toggleDrawer('left', false)}>
                    <div
                        tabIndex={0}
                        role="button"
                        onClick={this.toggleDrawer('left', false)}
                        onKeyDown={this.toggleDrawer('left', false)}>
                        {sideList}
                    </div>
                </Drawer>

            </div>
        );
    }


}
const mapStateToProps = (state, ownProps) => {
    return {
        state: state,
        ownProps: ownProps
    }
};

const mapDispatchToProps = {

};
export default compose(
    withRouter,
    connect(mapStateToProps, mapDispatchToProps)
)(NavigationMenu)
