// containers/phones/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import R from 'ramda'
import {Link} from 'react-router'

import {
  fetchPhones,
  loadMorePhones,
  addPhoneToBasket,
  fetchCategories,
  fetchUsers
} from 'actions'
import {getPhones} from 'selectors'

class Phones extends Component {
  componentDidMount () {
    this.props.fetchPhones();
    this.props.fetchCategories();
    this.props.fetchUsers();
  }

  renderPhone (phone, index) {
    console.log(this.props);
    const {addPhoneToBasket} = this.props;
    const shortDescription = `${R.take(60, phone.description)}...`;

    return (
      <div className='col-sm-4 col-lg-4 col-md-4 book-list' key={index}>
        <div className='thumbnail'>
          <img
            className='img-thumbnail'
            src={phone.image}
            alt={phone.name}
          />
          <div className='caption'>
            <h4 className='pull-right'>${phone.price}</h4>
            <h4>
              <Link to={`/phones/${phone.id}`}>
                {phone.name}
              </Link>
            </h4>
            <p>{shortDescription}</p>
            <p className='itemButton'>
              <button
                className='btn btn-primary'
                onClick={() => addPhoneToBasket(phone.id)}
              >
                Buy Now!
              </button>
              <Link
                to={`/phones/${phone.id}`}
                className='btn btn-default'
              >
                More info
              </Link>
            </p>
          </div>
        </div>
      </div>
    )
  }


  render () {
      let listItems = [];
      console.log('this. hello');
    console.log(this.props.state.users);
    const {phones, loadMorePhones} = this.props;
    if(this.props.state.users.length  > 0){
       listItems = this.props.state.users.map((user) =>
            <li key={user._id}>
                <span> id - {user._id}</span>
                <span> role - {user.role}</span>
            </li>
        );
    }

    return (
      <div>
          <ul>{listItems}</ul>
        <div className='books row'>
          {phones.map((phone, index) => this.renderPhone(phone, index))}
        </div>
        <div className='row'>
          <div className='col-md-12'>
            <button
              onClick={loadMorePhones}
              className='pull-right btn btn-primary'
            >
              Load More
            </button>
          </div>
        </div>
      </div>

    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    phones: getPhones(state, ownProps),
    state: state,
    ownProps: ownProps
  }
}

const mapDispatchToProps = {
  fetchPhones,
  loadMorePhones,
  addPhoneToBasket,
  fetchCategories,
  fetchUsers
}

export default connect(mapStateToProps, mapDispatchToProps)(Phones)
